x/merkledb/db.go

```diff
@@ -1250,6 +1258,43 @@ func (db *merkleDB) getNode(key Key, hasValue bool) (*node, error) {
 	return db.intermediateNodeDB.Get(key)
 }
 
+func (db *merkleDB) Clear() error {
+	db.commitLock.Lock()
+	defer db.commitLock.Unlock()
+
+	db.lock.Lock()
+	defer db.lock.Unlock()
+
+	// Clear caches
+	if err := db.valueNodeDB.Clear(); err != nil {
+		return err
+	}
+	if err := db.intermediateNodeDB.Clear(); err != nil {
+		return err
+	}
+
+	// Clear nodes from disk
+	if err := database.ClearPrefix(db.baseDB, valueNodePrefix, clearBatchSize); err != nil {
```

Comment:
Are these needed?  isn't the db cleared in db.valueNodeDB.Clear() and db.intermediateNodeDB.Clear() already?